packages/gitbook/src/components/Announcement/AnnouncementDismissedScript.tsx

```diff
@@ -0,0 +1,63 @@
+'use client';
+
+import {
+    ANNOUNCEMENT_CSS_CLASS,
+    ANNOUNCEMENT_DAYS_TILL_RESET,
+    ANNOUNCEMENT_STORAGE_KEY,
+} from './constants';
+
+/**
+ * Inject a script to read the local storage state for the announcement banner and apply the appropriate CSS class to the <html> element as early as possible.
+ * Bypasses react state to prevent flickering.
+ */
+export function AnnouncementDismissedScript() {
+    /**
+     * Read the local storage state for the announcement banner and apply the appropriate CSS class to the <html> element.
+     */
+    function checkStorageForDismissedScript(
```

Comment:
You can move this function out of the component as it's constant.

Usually I'd recommend having it in another script to make sure we don't have side effects of some imports.